.github/workflows/terraform.yml

```diff
@@ -0,0 +1,39 @@
+name: 'Terraform'
+
+on:
+  pull_request:
+    branches:
+      - master
+
+env:
+  TF_VERSION: 0.14.3
+
+jobs:
+  terraform:
+    name: 'terraform'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: ${{ env.TF_VERSION }}
+
+      - name: Setup providers configuration
+        shell: bash
+        run: |
+          cat << EOF > provider.tf
+          provider "azurerm" {
+            features {}
+          }
+          EOF
```

Comment:
use the example folder instead for the init and validate then you won't need this

(fmt will still need to be run from root)

Reply:
Using the example folder for the init and validate as advised